src/DeletableList/DeletableList.types.ts

```diff
@@ -0,0 +1,10 @@
+export interface DeletableListProps {
+  /**
+   * Array of list to be displayed.
+   */
+  list: string[];
```

Comment:
For consistency with the TransferList component I would recommend to change this prop name to `items`.  I think it makes more sense to pass items in to the list component rather than a list in to the list component.

```suggestion
  /**
   * Array of list items to be displayed.
   */
  items: string[];
```